apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/RequestResponsePopulateMdcFilterTest.java

```diff
@@ -0,0 +1,180 @@
+package gov.cms.bfd.server.war;
+
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import gov.cms.bfd.server.sharedutils.BfdMDC;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.slf4j.spi.MDCAdapter;
+
+/** Unit tests for {@link BfdMDC}. */
+@ExtendWith(MockitoExtension.class)
+public final class RequestResponsePopulateMdcFilterTest {
+  /** A mocked MDC adapter for use in the tests. */
+  @Mock MDCAdapter mdcMock;
+
+  /** A mocked servlet request for use in the tests. */
+  @Mock HttpServletRequest servletRequest;
+
+  /** A mocked servlet response for use in the tests. */
+  @Mock HttpServletResponse servletResponse;
+
+  /** A mocked filter chain for use in the tests. */
+  @Mock FilterChain filterChain;
+
+  /** The previous MDC adapter. */
+  private static MDCAdapter previousAdapter;
+
+  /** Set up tests with a mocked MDC adapter and make sure the MDC context is clear. */
+  @BeforeEach
+  public void beforeEach() {
+    mdcMock.clear();
+    BfdMDC.setMDCAdapter(mdcMock);
+    Mockito.reset(mdcMock);
+  }
+
+  /** Gets the previous MDC adapter. */
+  @BeforeAll
+  public static void beforeAll() {
+    previousAdapter = BfdMDC.getMDCAdapter();
+  }
+
+  /** Restores the MDC adapter. */
+  @AfterAll
+  public static void afterAll() {
+    BfdMDC.setMDCAdapter(previousAdapter);
+  }
+
+  /***
+   * Test arguments.
+   * @return arguments
+   */
+  public static Stream<Arguments> loggingTest() {
+    return Stream.of(
+        arguments("nonsensitive", "nonsensitive"),
```

Comment:
If anyone has ideas for additional test cases to add here, I'd be happy to include them.